WordPress/gutenberg
# Move legacy-widget block over to edit-widgets package

## Description
This PR ensures that the legacy-widget block is only registered in the widgets editor as discussed in https://github.com/WordPress/gutenberg/issues/24900. While this may change at some point in the future, as of now it's unclear how it could work in the post editor.

## How has this been tested?
1. Go to post editor, confirm that legacy widget block is not available there
1. Go to widgets editor, confirm that legacy widget block is available there

## Types of changes
Non-breaking change

## Checklist:
